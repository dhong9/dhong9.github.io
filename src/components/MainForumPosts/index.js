import Grid from "@mui/material/Grid";
import Typography from "@mui/material/Typography";
import Divider from "@mui/material/Divider";
import MKBox from "components/MKBox";

import ReactMarkdown from "react-markdown";

// prop-types is a library for typechecking of props
import PropTypes from "prop-types";

export default function MainForumPosts(props) {
  const { posts, title } = props;

  return (
    <Grid
      item
      xs={12}
      md={8}
      sx={{
        "& .markdown": {
          py: 3,
        },
      }}
    >
      <Typography variant="h6" gutterBottom>
        {title}
      </Typography>
      <Divider />
      {posts.map(({ id, isPlainText, body }) =>
        isPlainText ? (
          <MKBox p={3} key={id}>
            {body}
          </MKBox>
        ) : (
          <ReactMarkdown className="markdown" key={id}>
            {body}
          </ReactMarkdown>
        )
      )}
    </Grid>
  );
}

// Typechecking props for the BackgroundBlogCard
MainForumPosts.propTypes = {
  posts: PropTypes.arrayOf(PropTypes.oneOfType([PropTypes.object])),
  title: PropTypes.string.isRequired,
};
