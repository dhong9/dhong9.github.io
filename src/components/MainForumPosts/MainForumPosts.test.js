// Unit test libraries
import renderer from "react-test-renderer";

// Component to test
import MainForumPosts from "components/MainForumPosts";

describe("MainForumPosts", () => {
  it("renders", () => {

    const posts = []
    const title = "Test Post"

    const component = renderer.create(<MainForumPosts posts={posts} title={title} />);

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
