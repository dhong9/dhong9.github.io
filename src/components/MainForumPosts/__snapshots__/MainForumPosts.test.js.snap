// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainForumPosts renders 1`] = `
<div
  className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-8 css-sjz3wq-MuiGrid-root"
>
  <h6
    className="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom css-18k87ye-MuiTypography-root"
  >
    Test Post
  </h6>
  <hr
    className="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
  />
</div>
`;
